homeassistant/components/switcher_kis/__init__.py

```diff
@@ -0,0 +1,268 @@
+"""Home Assistant Switcher Component.
+
+For controlling the Switcher Boiler Device (https://www.switcher.co.il/).
+Please follow configuring instructions here:
+    https://www.home-assistant.io/components/switcher_kis/
+
+Author: Tomer Figenblat
+"""
+
+from asyncio import TimeoutError as Asyncio_TimeoutError, wait_for
+from datetime import datetime, timedelta
+from functools import partial
+from logging import getLogger
+from typing import Dict, Optional
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.const import CONF_ICON, CONF_NAME, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import Event, HomeAssistant, ServiceCall
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import (async_listen_platform,
+                                             async_load_platform)
+from homeassistant.helpers.event import async_call_later
+
+_LOGGER = getLogger(__name__)
+
+REQUIREMENTS = ['aioswitcher==2019.3.21']
+DOMAIN = 'switcher_kis'
+ENTITY_ID_FORMAT = DOMAIN + '_{}'
+
+CONF_AUTO_OFF = 'auto_off'
+CONF_DAYS = 'days'
+CONF_DEVICE_ID = 'device_id'
+CONF_DEVICE_PASSWORD = 'device_password'
+CONF_END_TIME = 'end_time'
+CONF_INCLUDE_SCHEDULE_SENSORS = 'include_schedule_sensors'
+CONF_PHONE_ID = 'phone_id'
+CONF_RECURRING = 'recurring'
+CONF_SCHEDULE_ID = 'schedule_id'
+CONF_SCHEDULE_SCAN_INTERVAL = 'schedules_scan_interval'
```

Comment:
Can we just use the normal scan_interval here?

Reply:
@rohankapoorcom 
I'm afraid it will be confusing for the users.
The component is divided to two platforms:
- A `switch` entity which represents the device itself including its state, the `switch` platform is of iot class `local push` and has no need for any scan/update interval configured.
- 8 `sensor` entities representing the 8 schedule slots allowed on the device, the `sensor` platform is of iot class `local pull` and needs the scan/update interval configured.

So it needs to be clear to the user the the `schedules_scan_interval` configuration value refers to the schedules data retrieval (the `sensor` platform) and not the state retrieval (the `switch` platform).

Reply:
Removed, not relevant anymore.